File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = defaultdict(int)
        for num in nums:
            freq[num] += 1
        n = len(freq)
        res = []
        keys = list(freq.keys())
        values = list(freq.values())
        for i in range(k):
            res.append((values[i],keys[i]))
        heapq.heapify(res)
        for i in range(k,n):
            if values[i] > res[0][0]:
                heapq.heappop(res)
                heapq.heappush(res,(values[i],keys[i]))
        return [x[1] for x in res]


        

File: Data Structures & Algorithms/count-paths/submission-1.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        vector<int> dp(n+1, 1);
        dp[0] = 0;
        for(int i = 2; i <= m; i++){
            for(int j = 1; j <= n; j++){
                dp[j] += dp[j-1];
            }
        }
        return dp[n];
    }
};
// 1,1 = 1
// 1,2 = 1
// 2,1 = 1
// 2,2 = 1
// 2,3 = 3
// 3,2 = 3
// 3,3 